src/test/scala/REPLActionTest.scala

```diff
@@ -0,0 +1,73 @@
+import com.intellij.ide.DataManager
+import com.intellij.mock.MockVirtualFile.{dir, file}
+import com.intellij.openapi.actionSystem.{AnAction, AnActionEvent, Presentation}
+import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import fi.aalto.cs.apluscourses.intellij.actions.REPLAction
+import org.jetbrains.plugins.scala.console.configuration.{ScalaConsoleConfigurationType, ScalaConsoleRunConfiguration, ScalaConsoleRunConfigurationFactory}
+import org.junit.Assert._
+import org.junit.Test
+import org.mockito.Mockito.{spy, times, verify}
+
+class REPLActionTest extends BasePlatformTestCase {
+
+  @Test
+  def testCheckFileOrFolderWithCorrectFileInputWorks(): Unit = {
+    val nonNullFileOrFolder = dir("directory")
+    val action = new REPLAction
+
+    assertFalse("Returns 'true' if the given folder is 'null'", action.checkFileOrFolderIsNull(nonNullFileOrFolder))
+  }
+
+  @Test
+  def testCheckFileOrFolderWithCorrectFolderInputWorks(): Unit = {
+    val nonNullFileOrFolder = file("file")
+    val action = new REPLAction
+
+    assertFalse("Returns 'true' if the given file is 'null'", action.checkFileOrFolderIsNull(nonNullFileOrFolder))
+  }
+
+  @Test
+  def testCheckFileOrFolderWithEmptyInputFails(): Unit = {
+    val nonNullFileOrFolder = null
+    val action = new REPLAction
+
+    assertTrue("Returns 'false' if the given file/folder is not 'null'", action.checkFileOrFolderIsNull(nonNullFileOrFolder))
+  }
+
+  @Test
+  def testAdjustRunConfigurationSettingsWithValidInputWorks(): Unit = {
+    val project = getProject
+    val module = getModule
+    val factory = new ScalaConsoleRunConfigurationFactory(new ScalaConsoleConfigurationType)
+    val configuration = factory.createTemplateConfiguration(project).asInstanceOf[ScalaConsoleRunConfiguration]
+    configuration.setWorkingDirectory("fakeDirectoryName")
+    val action = new REPLAction
+
+    action.adjustRunConfigurationSettings(module, configuration)
+
+    assertEquals("Customized title of the Scala REPL contains the name of the module started from", "Scala REPL for module: light_idea_test_case", configuration.getName)
+    assertTrue("Customized working directory is set to be a subFolder of '/tmp/'", configuration.getWorkingDirectory.contains("/tmp/unitTest_"))
+    assertTrue("Configuration is triggered with the module is has been started from", configuration.getAllModules.contains(module))
+  }
+
+  @Test(expected = classOf[IllegalArgumentException])
+  def testCustomDoRunActionIsTriggered(): Unit = {
+    //given
+    val action = new REPLAction
+    val spyAction = spy(action)
+    val actionEvent = createActionEvent(action)
+
+    //when
+    spyAction.actionPerformed(actionEvent)
+
+    //then
+    verify(spyAction, times(1)).customDoRunAction(actionEvent)
```

Comment:
actually, testing the `customDoRunAction(...)` would be more relevant. I'm still thinking of how to approach this

Reply:
The last two lines of `customDoRunAction(...)` function are those that *do* something; all before that is setup, right? Perhaps, we could extract those two lines behind and interface which `customDoRunAction(...)` would then call like:
```
consoleFacade.showConsole(module, setting, whatever);
```
and in unit-tests, it is verified that the method is called with correct parameters.

Okay, but there would still be a lot of nasty mocking, so perhaps we leave that method out of testing; also it contains only a little logic so that's fine, I guess.

Reply:
I've considered the interface approach option as well (regardless of the way it works with Scala), thus it just so much mocking, that I would not maybe unit test the whole thing